utils/zerovec/derive/src/lib.rs

```diff
@@ -45,6 +51,39 @@ pub fn varule_derive(input: TokenStream) -> TokenStream {
 /// For enums, this implementation will generate a crate-public `fn new_from_u8(value: u8) -> Option<Self>`
 /// method on the main type that allows one to construct the value from a u8. If this method is desired
 /// to be more public, it should be wrapped.
+///
+/// # Example
+///
+/// ```rust
+/// use zerovec::ZeroVec;
+///
+/// #[zerovec::make_ule(DateULE)]
+/// #[derive(Copy, Clone, PartialEq, Eq, Ord, PartialOrd, serde::Serialize, serde::Deserialize)]
+/// struct Date {
+///     y: u64,
+///     m: u8,
+///     d: u8
+/// }
+///
+/// #[derive(serde::Serialize, serde::Deserialize)]
+/// struct Dates<'a> {
+///     #[serde(borrow)]
+///     dates: ZeroVec<'a, Date>   
+/// }
+///
+/// let dates = Dates { dates: ZeroVec::alloc_from_slice(&[Date {y: 1985, m: 9, d: 3}, Date {y: 1970, m: 2, d: 20}, Date {y: 1990, m: 6, d: 13}]) };
+///
+/// let postcard_bytes = postcard::to_stdvec(&dates)
```

Comment:
I feel like we should either consistently use bincode or postcard in the docs. Thoughts?

I like postcard as a crate, but `bincode`  uses `serialize`/`deserialize` terminology so it's clearer what it's doing.

Reply:
Postcard feels more niche. For a utility crate like this I think bincode is better because more people will be familiar with it. For ICU4X componentes we should use postcard though, as it's the ICU4X recommended scheme.